apollo-router/src/context/mod.rs

```diff
@@ -70,6 +70,13 @@ impl Context {
 }
 
 impl Context {
+    pub(crate) fn operation_name(&self) -> Option<String> {
+        // This method should be removed once we have a proper way to get the operation name.
```

Comment:
https://github.com/apollographql/router/blob/dd7bac0e9269fbe8c1b3be431cf9e4373862d084/apollo-router/src/services/layers/query_analysis.rs#L111 this looks to me like the right way to get the operation name (starting at the supergraph_service)